packages/core/src/components/forms/_numeric-input.scss

```diff
@@ -0,0 +1,234 @@
+`value` can be provided as either a number or a string.
+
+The combined support of arbitrary text entry, controlled mode, and custom
+callbacks makes it possible to extend the numeric input's basic functionality in
+powerful ways. As shown in the example above, one could extend the numeric input
+component with support for mathematical expressions as follows:
+
+```
+import { NumericInput } from BlueprintComponents;
+import * as SomeLibrary from "some-library";
+
+export class NumericInputExample extends React.Component<{}, { value?: number |
+string }> {
+    public state = { value: null };
+
+    public render() {
+        return (
+            <NumericInput
+                onKeyDown={this.handleKeyDown}
+                value={this.state.value}
+            />
+        );
+    }
+
+    private handleKeyDown = (e: React.FormEvent<HTMLInputElement>) {
+        const value = e.target.value;
+        const result = SomeLibrary.evaluateMathExpression(value);
+        this.setState({ value: result });
+    }
+}
+```
+
+Styleguide components.forms.numeric-input.js.controlled
+*/
+
+$pt-numeric-input-z-index-below: 1;
+$pt-numeric-input-z-index-above: 2;
+
+.pt-numeric-input {
+
+  // we need a super-specific selector here to override very-specific selectors
+  // defined elsewhere. -.-
+  .pt-button-group.pt-vertical > .pt-button {
+    @include pt-button-height($pt-button-height);
+
+    // change to display: block to avoid a weird, jagged focus outline due to
+    // the reduced line-height.
+    display: block;
+    min-height: $pt-button-height / 2;
+    line-height: $pt-button-height / 2 - 1px;
+
+    &:first-child {
+      border-radius: 0 $pt-border-radius 0 0;
+      height: ($pt-button-height / 2) + 1;
+    }
+
+    &:last-child {
+      border-radius: 0 0 $pt-border-radius 0;
+      height: $pt-button-height / 2;
+    }
+
+    &[class*="pt-icon-"]::before {
+      // these styles are super unintuitive, but they basically are present to
+      // ensure that the focus outlines around the buttons will be perfect
+      // rectangles. delete any one of these styles, and the focus outlines will
+      // become jagged, because we're trying to fit a 16-px tall icon inside a
+      // ~15px-tall button.
+      display: block;
+      height: $pt-button-height / 2 - 1;
+      overflow: hidden;
+      line-height: $pt-button-height / 2 - 1;
+    }
+  }
+
+  // fix button border radii when the buttons are on the left
+  .pt-button-group:first-child.pt-vertical > .pt-button {
+    &:first-child {
+      border-radius: $pt-border-radius 0 0 0;
+    }
+
+    &:last-child {
+      border-radius: 0 0 0 $pt-border-radius;
+    }
+  }
+
+  // lift the focused group to the top so that its focus outline renders above
+  // everything else (you'd think this would happen for free by using the
+  // pt-control-group, but the fact that we're nesting groups within groups
+  // breaks that assumption).
+
+  &.pt-numeric-input-button-group-focused {
```

Comment:
Do we still need that with the recent button refactor? If yes, hopefully we can address in input/control group directly. That class is scary 😨